app/services/scenes/scenes.ts

```diff
@@ -227,13 +232,16 @@ export class ScenesService extends StatefulService<IScenesState> {
     this.scenes.forEach(scene => scene.setLockOnAllItems(locked));
   }
 
-  getSourceScenes(sourceId: string): Scene[] {
-    const resultScenes: Scene[] = [];
+  getSourceItemCount(sourceId: string): number {
+    let count = 0;
+
     this.scenes.forEach(scene => {
-      const items = scene.getItems().filter(sceneItem => sceneItem.sourceId === sourceId);
-      if (items.length > 0) resultScenes.push(scene);
+      scene.getItems().forEach(sceneItem => {
+        if (sceneItem.sourceId === sourceId) count += 1;
+      });
     });
-    return resultScenes;
+
+    return count;
```

Comment:
i feel like a double filter would be more effective here

Reply:
Unless I'm missing something, I think the outer iteration would need to be a reduce operation.  This seemed quick and obvious, but I suppose definitely less functional.

Reply:
ah wait yea i misread this functionality nevermind